apps/nestjs-wallet/src/vc-api/exchanges/exchange.service.ts

```diff
@@ -127,8 +129,15 @@ export class ExchangeService {
         errors: [`${transactionId}: verification of presentation proof not successful`, ...result.errors]
       };
     }
-    const response = transaction.processPresentation(verifiablePresentation);
+    const { response, callback } = transaction.processPresentation(verifiablePresentation);
     await this.transactionRepository.save(transaction);
+    callback?.forEach((callback) => {
+      try {
+        this.httpService.post(callback.url, response);
```

Comment:
Don't forget to wait for the error. `post` method should return an Observable

Reply:
Do you mean subscribe to the Observable and log an error if it occurs?